File: Low Level Design (LLD)/DesignPatterns/3. Structural/Command/Java/Main.java
public class Main {
    public static void main(String[] args) {
        // Receivers
        Light light = new Light();
        Fan fan = new Fan();

        // Commands
        Command lightOn = new LightOnCommand(light);
        Command lightOff = new LightOffCommand(light);
        Command fanOn = new FanOnCommand(fan);
        Command fanOff = new FanOffCommand(fan);

        // Invoker
        RemoteControl remote = new RemoteControl();

        // Execute commands
        remote.setCommand(lightOn);
        remote.pressButton();

        remote.setCommand(fanOn);
        remote.pressButton();

        remote.setCommand(lightOff);
        remote.pressButton();

        remote.setCommand(fanOff);
        remote.pressButton();
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Command/Cpp/main.cpp
#include "include/Light.h"
#include "include/Fan.h"
#include "include/LightOnCommand.h"
#include "include/LightOffCommand.h"
#include "include/FanOnCommand.h"
#include "include/FanOffCommand.h"
#include "include/RemoteControl.h"

int main() {
    Light light;
    Fan fan;

    LightOnCommand lightOn(&light);
    LightOffCommand lightOff(&light);
    FanOnCommand fanOn(&fan);
    FanOffCommand fanOff(&fan);

    RemoteControl remote;

    remote.setCommand(&lightOn);
    remote.pressButton();

    remote.setCommand(&fanOn);
    remote.pressButton();

    remote.setCommand(&lightOff);
    remote.pressButton();

    remote.setCommand(&fanOff);
    remote.pressButton();

    return 0;
}
